### DISCUSSION TITLE
Different Result in Editor than when rendering qmd

### DISCUSSION THREAD

**Discussion by Beedmoser** [quarto-dev/quarto-cli#10848] [Q&A] [answered]
### Description

I can run the below code in the editor and get the desired results. However, I cannot get the same results when I try to render the same code.

---
title: "mwe"
format: html
---

```{r}
    library(terra)
    library(rnaturalearth)
    
    france <- rnaturalearth::ne_states(country = "france",
                                       returnclass = "sv")
    plot(france)
```

 Then, Quarto throws:
 ==> quarto preview reprex.qmd --to html --no-watch-inputs --no-browse

processing file: reprex.qmd
  |...................................                 |  67% [unnamed-chunk-1]

Quitting from lines 7-13 [unnamed-chunk-1] (reprex.qmd)
Error in `value[[3L]]()`:
! Failed to install the rnaturalearthhires package.
  Please try installing the package for yourself using the following command: 
     devtools::install_github("ropensci/rnaturalearthhires")
Backtrace:
 1. rnaturalearth::ne_states(country = "france", returnclass = "sv")
 2. rnaturalearth::check_rnaturalearthhires()
 3. rnaturalearth::install_rnaturalearthhires()
 4. base::tryCatch(...)
 5. base (local) tryCatchList(expr, classes, parentenv, handlers)
 6. base (local) tryCatchOne(...)
 7. value[[3L]](cond)
                                                                                                            
Ausführung angehalten
![Screenshot 2024-09-20 180101](https://github.com/user-attachments/assets/0130c154-84e3-4e16-9ddb-66aeb135c7c4)

**Comment by mcanouil** [comment]
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.

This being said, it works "fine" besides the fact that the R package `rnaturalearth` is installing `rnaturalearthhires` without any consent and regardless of your setup (in my opinion, that's bad).

You should check that you are running Quarto in the same environment where the R packages are installed.

**Comment by Beedmoser** [comment, marked as answer]
I withdraw the question. The solution is to insert .libPaths("C:/Users/{…}/AppData/Local/R/win-library/4.3") since it is possible to have different paths for quarto as for r.

Best

Beat


Beat Moser

Bergstrasse 9
8322 Madetswil

FON +41 79 257 96 00

Von: Mickaël Canouil ***@***.***>
Gesendet: Montag, 30. September 2024 13:09
An: quarto-dev/quarto-cli ***@***.***>
Cc: Beat Moser ***@***.***>; Author ***@***.***>
Betreff: Re: [quarto-dev/quarto-cli] Different Result in Editor than when rendering qmd (Discussion #10848)


Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, i.e., usually four ````.
See https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us.
